### DISCUSSION TITLE
Are there different methods to add models?

### DISCUSSION THREAD

**Discussion by kadirnar** [obss/sahi#643] [Q&A] [answered]
Merhabalar,

SAHI kütüphanesinin farklı algoritmalarla nasıl çalıştığını test etmek istiyorum. Model.py dosyasında işlem yapmadan gelen score,bbox,category_name, category_id değerlerini ObjectPrediction fonksiyonun içinde tanımlamak yeterli mi? 

[Dethub](https://github.com/kadirnar/dethub/blob/main/dethub/model.py#L56-L66) bu projede istenen değişkenleri bir sözlükte tutuyorum. En basit şekilde SAHI'ye nasıl entegre edebilirim?

**Comment by fcakyon** [comment]
Merhaba Kadir,

Farkli algoritmalardan kastettigin nedir, detaylandirabilir misin? Obje tespit modelleri mi?

Test etmek istedigin sey nedir bir de, biraz acabilir misin? Sliced inference sonuclari mi? Standard inference sonuclari mi? Evaluation sonuclari mi? Gorsellestirmeler mi?

**Comment by kadirnar** [comment]
SAHI'nin desteklemediği nesne tespit modelleri(Tfhub,NanoDet,scaled-yolov4...) ve takip algoritmalarını(DeepSort,ByteTrack,StrongSort...) test etmek istiyorum. Sliced inference kullandığımızda ne kadar stabil ve yüksek doğruluklu sonuç verdiğini test etmek istiyorum.

**Reply by yanqianshilu** [reply]
Hello, I used sahi slice reasoning in nanodet model to reduce the accuracy, may I ask if you have encountered similar problems?

**Reply by kadirnar** [reply]
I didn't use the Nanodet model.

**Comment by fcakyon** [comment, marked as answer]
SAHI'ye tepit modeli eklemenin tek yolu model.py icerisine yeni class eklemek. Tespit sonuclarini sonradan SAHI formatina cevirerek sliced inference almak mumkun degil malesef.

SAHI'nin uzerine tracker destegi eklemek icin, her frame uzerinde SAHI ile prediction aldiktan sonra, resulttaki object_prediction_list elemanlarini istedigin trackera girdi olarak verebilirsin.

**Reply by kadirnar** [reply]
@fcakyon SAHI reposuna tracker algoritmalarını eklemek ile ilgili düşünceleriniz nedir? Olumlu bakıyorsanız pip desteği oluşturup PR gönderebilirim. Veya kişisel repo açabilirim.

**Reply by fcakyon** [reply]
SAHI reposuna tracker desteği eklemeyi düşünmüyoruz bu noktada, geliştirme yükünü çok arttırır.

SAHI kullanarak geliştirilmiş tracker içeren projelerin linklerini SAHI reposuna ekleyebiliriz isteyenlerin kullanması için 👍

**Comment by kadirnar** [comment]
Cevabınız için teşekkür ederim.
